number_theory/modular.h

```diff
@@ -33,6 +36,20 @@ constexpr T normalize(T x, T modulus) {
   return y;
 }
 
+// Casts |number| to given type and returns the result.
+// Throws invalid_argument exception if the conversion does overflow/underflow.
+template <typename T, typename U>
+constexpr T numeric_cast(U number) {
+  static_assert(
+      std::numeric_limits<T>::is_integer && std::numeric_limits<U>::is_integer,
+      "numeric_cast can be done only between integer types.");
+  if (!std::in_range<T>(number)) {
+    throw std::invalid_argument(
```

Comment:
Maybe `range_error` is a better exception? What do you think?

> It can be used to report range errors (that is, situations where a result of a computation cannot be represented by the destination type).

Reply:
Thanks for the suggestion, I agree that `range_error` suits better. I will change it accordingly.